Upgrade to spring cloud task 2.2.x

Simple version change in a pom results some compile errors which needs to get resolved either in task or dataflow.

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.0:testCompile (default-testCompile) on project spring-cloud-dataflow-server-core: Compilation failure: Compilation failure: 
[ERROR] /home/jvalkealahti/repos/spring/spring-cloud-dataflow/spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/configuration/JobDependencies.java:[203,71] constructor TaskRepositoryInitializer in class org.springframework.cloud.task.repository.support.TaskRepositoryInitializer cannot be applied to given types;
[ERROR]   required: org.springframework.cloud.task.configuration.TaskProperties
[ERROR]   found: no arguments
[ERROR]   reason: actual and formal argument lists differ in length
[ERROR] /home/jvalkealahti/repos/spring/spring-cloud-dataflow/spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/configuration/TaskServiceDependencies.java:[137,71] constructor TaskRepositoryInitializer in class org.springframework.cloud.task.repository.support.TaskRepositoryInitializer cannot be applied to given types;
[ERROR]   required: org.springframework.cloud.task.configuration.TaskProperties
[ERROR]   found: no arguments
[ERROR]   reason: actual and formal argument lists differ in length

```

Not sure if there's any need for `TaskRepositoryInitializer` in a test configs as flyway manages a schema.

--- a/pom.xml
+++ b/pom.xml
@@ -73,7 +73,9 @@
 
 		<spring-cloud-skipper.version>2.2.0.BUILD-SNAPSHOT</spring-cloud-skipper.version>
 
-		<spring-cloud-task.version>2.1.3.BUILD-SNAPSHOT</spring-cloud-task.version>
+		<spring-cloud-task.version>2.2.0.BUILD-SNAPSHOT</spring-cloud-task.version>
+
+		<spring-security-oauth-autoconfigure.version>2.2.0.M4</spring-security-oauth-autoconfigure.version>
 
 		<!-- Note: Make sure to update `spring-cloud-build` version and `spring-cloud-dependencies` version (spring-cloud.version)
 			 are in sync with the https://github.com/spring-cloud/spring-cloud-release/blob/master/pom.xml updates for the respective
@@ -234,7 +236,7 @@
 			<dependency>
 				<groupId>org.springframework.security.oauth.boot</groupId>
 				<artifactId>spring-security-oauth2-autoconfigure</artifactId>
-				<version>${spring-boot.version}</version>
+				<version>${spring-security-oauth-autoconfigure.version}</version>
 			</dependency>
 			<dependency>
 				<groupId>io.fabric8</groupId>

--- a/spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/configuration/JobDependencies.java
+++ b/spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/configuration/JobDependencies.java
@@ -93,7 +93,6 @@ import org.springframework.cloud.task.repository.TaskRepository;
 import org.springframework.cloud.task.repository.support.SimpleTaskExplorer;
 import org.springframework.cloud.task.repository.support.SimpleTaskRepository;
 import org.springframework.cloud.task.repository.support.TaskExecutionDaoFactoryBean;
-import org.springframework.cloud.task.repository.support.TaskRepositoryInitializer;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.core.io.DefaultResourceLoader;
@@ -196,13 +195,6 @@ public class JobDependencies {
 	@Bean
 	public TaskLogsController taskLogsController(TaskExecutionService taskExecutionService) {
 		return new TaskLogsController(taskExecutionService);
-	}
-
-	@Bean
-	public TaskRepositoryInitializer taskExecutionRepository(DataSource dataSource) {
-		TaskRepositoryInitializer taskRepositoryInitializer = new TaskRepositoryInitializer();
-		taskRepositoryInitializer.setDataSource(dataSource);
-		return taskRepositoryInitializer;
 	}
 
 	@Bean

--- a/spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/configuration/TaskServiceDependencies.java
+++ b/spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/configuration/TaskServiceDependencies.java
@@ -73,7 +73,6 @@ import org.springframework.cloud.task.repository.TaskRepository;
 import org.springframework.cloud.task.repository.support.SimpleTaskExplorer;
 import org.springframework.cloud.task.repository.support.SimpleTaskRepository;
 import org.springframework.cloud.task.repository.support.TaskExecutionDaoFactoryBean;
-import org.springframework.cloud.task.repository.support.TaskRepositoryInitializer;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.context.annotation.Import;
@@ -131,13 +130,6 @@ public class TaskServiceDependencies extends WebMvcConfigurationSupport {
 
 	@Autowired
 	DockerValidatorProperties dockerValidatorProperties;
-
-	@Bean
-	public TaskRepositoryInitializer taskExecutionRepository(DataSource dataSource) {
-		TaskRepositoryInitializer taskRepositoryInitializer = new TaskRepositoryInitializer();
-		taskRepositoryInitializer.setDataSource(dataSource);
-		return taskRepositoryInitializer;
-	}
 
 	@Bean
 	public TaskValidationService taskValidationService(AppRegistryService appRegistry,

